app/controllers/settings_controller.rb

```diff
@@ -0,0 +1,34 @@
+class SettingsController < ApplicationController
+  before_action :require_current_user
+
+  STUDENT_ID_PATTERN = /7\d{7}/
+
+  def edit_profile
+  end
+
+  def update_profile
```

Comment:
Assignment Branch Condition size for update_profile is too high. [27.24/20]<br>Method has too many lines. [19/10]